strongbox-commons/src/main/java/org/carlspring/strongbox/io/StreamUtils.java

```diff
@@ -13,30 +18,68 @@ private StreamUtils()
 
     }
 
-    public static long getLength(ArtifactInputStream is)
-            throws IOException
+    public static long getLength(ByteRangeInputStream bris)
+        throws IOException
     {
-        InputStream target = is.getTarget();
-        if (!(target instanceof ByteRangeInputStream))
-        {
-            return -1L;
-        }
-        ByteRangeInputStream bris = (ByteRangeInputStream) target;
         return bris.getLength();
     }
 
-    public static void setCurrentByteRange(ArtifactInputStream is,
+    public static void setCurrentByteRange(ByteRangeInputStream bris,
                                            ByteRange byteRange)
-            throws IOException
+        throws IOException
     {
-        InputStream target = is.getTarget();
-        if (!(target instanceof ByteRangeInputStream))
-        {
-            return;
-        }
-        ByteRangeInputStream bris = (ByteRangeInputStream) target;
         bris.setCurrentByteRange(byteRange);
         bris.skip(byteRange.getOffset());
     }
 
+    public static <T extends InputStream> T findSource(Class<T> sourceClass,
```

Comment:
Do we really need the reflection in these two methods? Isn't there another way to do this?

Reply:
yeah this is not the best solution, but I wasn't able to find something better
I suggest to leave it as is until we find something better